src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs

```diff
@@ -15,9 +15,9 @@
 using Nerdbank.Streams;
 using Roslyn.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
 {
-    internal abstract class AbstractLanguageServerClient : ILanguageClient
+    internal abstract class AbstractInProcLanguageClient : ILanguageClient
```

Comment:
i don't like ILanguageClient.  However, if htat's the VS name, then i'd prefer our types use the same naming so it's clear the connection between the two.

Reply:
i added 'InProc' to the name to help convey where this client will sit.

Reply:
I think the type name should be `AbstractInProcLanguageServer` or `AbstractInProcLSPServer`, and not have "client" in the name. I think the platforms `ILanguageClient` this has to implement is named badly, and we shouldn't inherit that.

Reply:
while i don't love calling this XXXLangClient, i prefer that nam here because i can say "ah, tehse are the ones that are providing ILangClient to VS.   If we call this XXXLanguageServer *and* we then also have a type like InProcLanguageServer (which *does* exist), then it's just massively confusing.  i.e. using your name we would have *both*

AbstractInProcLanguageService as well as InProcLanguageServer, and tehre would be no relation between these two types! :(

With the formalization here we have AbstractInProcLanguageClient (and a few subclasses that follow that name), and then InProcLanguageServer. The names clearly indicate how their purposes and how they are unrelated.  Namely that the LangClient components are about plugging into that side, wihle InProcLangServer is the actual server piece that does all that heavy lifting of queuing, handler dispatching, etc.

this is why i wanted to do the renames here.  it wasjust so confusing to have names like ServerClient (wha...?) and then have multiple things with similar names that were actually unrelated.

Reply:
Yes, I see your point. Definitely having the two is confusing, and I can't think of a better name for `InProcLanguageServer`.